src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -60,9 +92,12 @@ internal static class DependentTypeFinder
             IImmutableSet<Project> projects,
             CancellationToken cancellationToken)
         {
-            return FindDerivedClassesAsync(
-                SymbolAndProjectId.Create(type, projectId: null), solution, projects,
-                transitive: true, cancellationToken: cancellationToken);
+            return FindTypesFromCacheOrComputeAsync(
+                type, solution, s_typeToTransitivelyDerivedClassesMap,
+                c => FindDerivedClassesAsync(
+                    SymbolAndProjectId.Create(type, projectId: null), solution, projects,
```

Comment:
❕ Due to caching, this method will "remember" the set of projects it gets called with the first time, and future callers with the same parameters will have their projects set ignored.